### PR TITLE
chore(codebuild): update BuildImage classes

### PR DESCRIPTION
**Pull request by nmussy**
### Issue # (if applicable)

None as far as I could tell

Overlaps with #29580

### Reason for this change

Align the CodeBuild images listed by the CDK with those offered by the SDK

### Description of changes

* Added missing build images
  * Windows 2022 was added with a warning, see [PR comment](https://github.com/aws/aws-cdk/pull/29616#issuecomment-2019991742)
* Removed `@deprecated` messages, see #29580
* Updated JSDoc comments to indicate AL2023 based images, see [docs](https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-available.html)
  * It might be a good idea to deprecate and rename `AMAZON_LINUX_2_STANDARD_3_0` to `AMAZON_LINUX_2023_STANDARD_3_0`, despite how the images are named. I'll leave it up to the maintainers
* Added `{@link}` tags where missing

### Description of how you validated changes

Using the [codebuild:ListCuratedEnvironmentImages](https://docs.aws.amazon.com/codebuild/latest/APIReference/API_ListCuratedEnvironmentImages.html) API command, and comparing it to the CDK.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nmussy** [comment]
Switching to a draft for now, I need to check whether or not `WINDOWS_SERVER_2022_CONTAINER` is available, the [docs aren't super clear about that](https://docs.aws.amazon.com/search/doc-search.html?searchPath=documentation-guide&searchQuery=WINDOWS_SERVER_2022_CONTAINER&this_doc_product=AWS%20CloudFormation&this_doc_guide=User%20Guide).

If it is, I'll probably update Windows in another PR to add an integ test

**Comment by nmussy** [comment]
As the [Cfn docs assert](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codebuild-project-environment.html#cfn-codebuild-project-environment-type), `WINDOWS_SERVER_2022_CONTAINER` is not a valid `Environment::Type`, sort of.

When attempting to deploy `aws/codebuild/windows-base:2022-1.0` with `"Type": "WINDOWS_SERVER_2022_CONTAINER"`, Cfn returns the following error:
```
Windows Server 2022 not supported for on-demand compute
```

I've also tried deploying it with the incorrect `Type`, i.e. `WINDOWS_SERVER_2019_CONTAINER`:
```
AWS CodeBuild curated image aws/codebuild/windows-base:2022-1.0 is not supported for projects with environment type WINDOWS_SERVER_2019_CONTAINER and compute type BUILD_GENERAL1_MEDIUM
```

This matches [Working with reserved capacity in AWS CodeBuild](https://docs.aws.amazon.com/codebuild/latest/userguide/fleets.html):
> On-demand ﬂeets do not support Windows Server 2022.

The CDK should be able to be set this up with [`CfnProject:ProjectFleetProperty`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_codebuild.CfnProject.ProjectFleetProperty.html), but that's going to be outside the scope of this PR.

I'll open an issue to resolve this later, but in the meantime I'll mark both `WindowsBuildImage.WIN_SERVER_CORE_2022_BASE_3_0` and `WindowsImageType.SERVER_2022` as `@deprecated`, explaining the situation. This can easily be updated later on.

File: packages/aws-cdk-lib/aws-codebuild/lib/project.ts [hunk 1]
```diff
@@ -1964,6 +1977,18 @@ export enum WindowsImageType {
    * The WINDOWS_SERVER_2019_CONTAINER environment type
    */
   SERVER_2019 = 'WINDOWS_SERVER_2019_CONTAINER',
+
+  /**
+   * The WINDOWS_SERVER_2022_CONTAINER environment type
+   *
+   * @deprecated Windows Server 2022 images are not supported for on-demand compute.
```
Comment (TheRealAmazonKendra) [inline packages/aws-cdk-lib/aws-codebuild/lib/project.ts:1984]: I don't think we should be adding version that we are immediately marking as deprecated.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Thanks for adding these. I'm not sure, however, that we should be adding types that we are also immediately marking as deprecated. Can you elaborate on your thinking here?

File: packages/aws-cdk-lib/aws-codebuild/lib/project.ts (hunk 1 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-codebuild/lib/project.ts:1984]: Yeah, I just want to give the user a warning that they probably won't be able to use the image as is. I thought the `@deprecated` was a good compromise. I can delete them here and we can add them back with #29617's PR

Comment (nmussy) [reply packages/aws-cdk-lib/aws-codebuild/lib/project.ts:1984]: @TheRealAmazonKendra Nudge

**Comment by nmussy** [comment]
Closing in favor of #29754, which will allow an integration test of Windows 2022

**Pull request closed without merging**